preview-src/timeline.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+
+import { Comment } from '../src/common/comment';
+import { TimelineEvent, isReviewEvent, isCommitEvent, isCommentEvent, isMergedEvent, isAssignEvent, ReviewEvent, CommitEvent, CommentEvent, MergedEvent, AssignEvent } from '../src/common/timelineEvent';
+import { commitIcon, mergeIcon } from './icon';
+import { Avatar, AuthorLink } from './user';
+import { groupBy } from '../src/common/utils';
+import { Spaced } from './space';
+import Timestamp from './timestamp';
+import { CommentView } from './comment';
+import Diff from './diff';
+
+export const Timeline = ({ events }: { events: TimelineEvent[] }) =>
+	<>{
+		events.map(event =>
+			// TODO: Maybe make TimelineEvent a tagged union type?
+			isCommitEvent(event)
+				? <CommitEventView key={event.id} {...event} />
+				:
+			isReviewEvent(event)
+				? <ReviewEventView key={event.id} {...event} />
+				:
+			isCommentEvent(event)
+				? <CommentEventView key={event.id} {...event} />
+				:
+			isMergedEvent(event)
+				? <MergedEventView key={event.id} {...event} />
+				:
+			isAssignEvent(event)
+				? <AssignEventView key={event.id} {...event} />
+				: null
+		)
+	}</>;
+
+export default Timeline;
+
+const CommitEventView = (event: CommitEvent) =>
+	<div className='comment-container commit'>
+		<div className='commit-message'>
+			{commitIcon}
+			<div className='avatar-container'>
+				<Avatar for={event.author} />
+			</div>
+			<AuthorLink for={event.author} />
+			<div className='message'>{event.message}</div>
+		</div>
+		<a className='sha' href={event.url}>{event.sha.slice(0, 7)}</a>
```

Comment:
I think this should be `event.htmlUrl` instead of `event.url`. We should get rid of the `url` property on `CommitEvent`

Reply:
Fixed.